### PR TITLE
feat: add '--tar' for 'kpm run' to run the kcl package tar.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #19 

#### 2. What is the scope of this PR (e.g. component or file name):

kpm run

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

'kpm run --tar' will extract tar into the temporary dir, compile the kcl program, and then delete the temporary dir.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/cmd/cmd_run.go [hunk 1]
```diff
@@ -35,60 +43,155 @@ func NewRunCmd() *cli.Command {
...
+					err = os.RemoveAll(destDir)
+					if err != nil {
+						return err
+					}
+				}
+				if compileErr != nil {
+					return compileErr
+				}
+				fmt.Print(compileResult)
+			} else { // If no tar packages specified by "--tar" to run
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:82]: fmt the code.

File: pkg/cmd/cmd_run.go [hunk 2]
```diff
@@ -35,60 +43,155 @@ func NewRunCmd() *cli.Command {
...
-				return err
-			}
+// runPkgInPath will load the 'KclPkg' from path 'pkgPath'.
+// And run the kcl package with entry file in 'entryFilePath' in 'vendorMode'.
+func runPkgInPath(pkgPath, entryFilePath string, vendorMode bool) (string, error) {
 
-			kclPkg.SetVendorMode(c.Bool("vendor"))
+	pkgPath, err := filepath.Abs(pkgPath)
+	if err != nil {
+		return "", errors.InternalBug
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:107]: errors.Wrap?

File: pkg/cmd/cmd_run.go [hunk 3]
```diff
@@ -35,60 +43,155 @@ func NewRunCmd() *cli.Command {
...
+	}
 
-			kpmHome, err := utils.GetAbsKpmHome()
-			if err != nil {
-				return err
-			}
+	kclPkg, err := pkg.LoadKclPkg(pkgPath)
+	if err != nil {
+		return "", errors.FailedToLoadPackage
```
Comment (Peefy) [inline pkg/cmd/cmd_run.go:112]: Why not use `return "", err`. or `return "", errors.Wrap(err)`?

Comment (zong-zhe) [reply pkg/cmd/cmd_run.go:113]: The number of code changes for this problem is relatively large for this PR, which will be adjusted later in the PR to fix https://github.com/KusionStack/kpm/issues/41 to optimize error message output.

File: pkg/cmd/cmd_run.go (hunk 2 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_run.go:108]: The number of code changes for this problem is relatively large for this PR, which will be adjusted later in the PR to fix https://github.com/KusionStack/kpm/issues/41 to optimize error message output.

File: pkg/cmd/cmd_run.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_run.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**